packages/api/tests/lib/test-cwSfExecutionEventUtils.js

```diff
@@ -0,0 +1,263 @@
+'use strict';
+
+const test = require('ava');
+const rewire = require('rewire');
+const cwSfExecutionEventUtils = rewire('../../lib/cwSfExecutionEventUtils');
+const {
+  getCumulusMessageFromExecutionEvent,
+  getFailedExecutionMessage
+} = cwSfExecutionEventUtils;
+
+test('getFailedExecutionMessage() returns the Cumulus message from the output of the last failed step', async (t) => {
+  const inputMessage = {
+    cumulus_meta: {
+      state_machine: 'arn:aws:states:us-east-1:111122223333:stateMachine:HelloWorld-StateMachine',
+      execution_name: 'my-execution-name'
+    }
+  };
+
+  const failedTaskOutput = { a: 1 };
+
+  const result = await cwSfExecutionEventUtils.__with__({
+    StepFunctions: {
+      getExecutionHistory: ({ executionArn }) => {
+        if (executionArn !== 'arn:aws:states:us-east-1:111122223333:execution:HelloWorld-StateMachine:my-execution-name') {
+          throw new Error(`Expected executionArn === 'arn:aws:states:us-east-1:111122223333:execution:HelloWorld-StateMachine:my-execution-name' but got ${executionArn}`);
+        }
+        return {
+          events: [
+            {
+              // lastStepFailedEvent
+              type: 'LambdaFunctionFailed',
+              id: 1
+            },
+            {
+              // failedStepExitedEvent
+              type: 'TaskStateExited',
+              previousEventId: 1,
+              stateExitedEventDetails: {
+                output: JSON.stringify(failedTaskOutput)
+              },
+              resource: 'x'
+            }
+          ]
+        };
+      }
+    }
+  })(() => getFailedExecutionMessage(inputMessage));
+
+  t.deepEqual(result, failedTaskOutput);
+});
+
+test('getFailedExecutionMessage() returns the input message if there is an error fetching the output of the last failed step', async (t) => {
+  const inputMessage = { a: 1 };
+
+  const actualResult = await getFailedExecutionMessage(inputMessage);
+
+  t.deepEqual(actualResult, inputMessage);
+});
+
+test.todo('getFailedExecutionMessage() does ?something? when no ActivityFailed or LambdaFunctionFailed events are found in the execution history');
```

Comment:
It should fall back to the input message in that case as well, I think?

Reply:
Fixed in 32adeb955